Framework/Algorithms/src/CompareWorkspaces.cpp

```diff
@@ -1340,27 +1339,36 @@ void CompareWorkspaces::recordMismatch(const std::string &msg, std::string ws1,
   m_result = false;
 }
 
+//------------------------------------------------------------------------------------------------
+/** Function which calculates absolute error between two values and analyses if
+this error is within the limits requested.
+
+@param x1    -- first value to check difference
+@param x2    -- second value to check difference
+@param atol  -- the tolerance of the comparison. Must be nonnegative
+
+@returns true if absolute difference is within the tolerance; false otherwise
+*/
+bool CompareWorkspaces::withinAbsoluteTolerance(double const x1, double const x2, double const atol) {
+  // NOTE !(|x1-x2| > atol) is not the same as |x1-x2| <= atol
+  return !(std::fabs(x1 - x2) > atol);
+}
+
 //------------------------------------------------------------------------------------------------
 /** Function which calculates relative error between two values and analyses if
-this error is within the limits
-* requested. When the absolute value of the difference is smaller then the value
-of the error requested,
-* absolute error is used instead of relative error.
+this error is within the limits requested.
 
-@param x1       -- first value to check difference
-@param x2       -- second value to check difference
-@param errorVal -- the value of the error, to check against. Should  be large
-then 0
+@param x1    -- first value to check difference
+@param x2    -- second value to check difference
+@param rtol  -- the tolerance of the comparison. Must be nonnegative
 
-@returns true if error or false if the value is within the limits requested
+@returns true if relative difference is within the tolerance; false otherwise
+@returns true if error or false if the relative value is within the limits requested
 */
-bool CompareWorkspaces::relErr(double x1, double x2, double errorVal) const {
-  double num = std::fabs(x1 - x2);
-  // how to treat x1<0 and x2 > 0 ?  probably this way
-  double den = 0.5 * (std::fabs(x1) + std::fabs(x2));
-  if (den < errorVal)
-    return (num > errorVal);
-
-  return (num / den > errorVal);
+bool CompareWorkspaces::withinRelativeTolerance(double const x1, double const x2, double const rtol) {
+  double const num = std::fabs(x1 - x2);
+  // create a standardized size to compare against
+  double const den = 0.5 * (std::fabs(x1) + std::fabs(x2));
```

Comment:
I think you need to do the `std::fabs` on the sum rather than the individuals. The current implementation has an issue when the two values are on either side of zero
```suggestion
  double const den = 0.5 * std::fabs(x1 + x2);
```

Reply:
I'll need to think about it more.  This method was there and I didn't want to change behavior too much. It also seems to be the standard way to treat opposite signs, unless we instead make it an error condition.

Suppose $x_1=1.0$ and $x_2=-1.0$.  Using sum of `std::fabs`, they differ by 2 and both have a magnitude of 1, so the relative difference is twice their value (2).

Using `std::fabs` on the sum would create a standard-size of 0 to compare against, and the result would be $+\infty$.

That might become a relevant for very tiny values oscillating near zero.